app/services/listing_index_service/search/zappy_adapter.rb

```diff
@@ -74,15 +72,18 @@ def format_params(original)
       }.or_else({})
 
       {
+       :'search[keywords]' => original[:keywords],
        :'page[number]' => original[:page],
        :'page[size]' => original[:per_page],
        :'filter[price_min]' => Maybe(original[:price_cents]).map{ |p| p.min }.or_else(nil),
        :'filter[price_max]' => Maybe(original[:price_cents]).map{ |p| p.max }.or_else(nil),
        :'filter[omit_closed]' => !original[:include_closed],
        :'filter[listing_shape_ids]' => Maybe(original[:listing_shape_ids]).join(",").or_else(nil),
        :'filter[category_ids]' => Maybe(original[:categories]).join(",").or_else(nil),
-       :'search[locale]' => original[:locale]
-      }.merge(search_params).merge(custom_fields).compact
+       :'search[locale]' => original[:locale],
+       :'filter[distance_max]' => original[:distance_max],
```

Comment:
I'd say defining this logically belongs together with the location_params above.